File: tres_flutter/fix_emails.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:firebase_auth/firebase_auth.dart';
import 'package:firebase_core/firebase_core.dart';

void main() async {
  // Initialize Firebase
  await Firebase.initializeApp();
  
  final firestore = FirebaseFirestore.instance;
  final auth = FirebaseAuth.instance;
  
  print('Fetching all users...');
  
  // Get all users from Firestore
  final usersSnapshot = await firestore.collection('users').get();
  
  print('Found ${usersSnapshot.docs.length} users\n');
  
  for (var doc in usersSnapshot.docs) {
    final data = doc.data();
    final uid = doc.id;
    
    print('User $uid:');
    print('  Current email in Firestore: "${data['email']}"');
    print('  Display name: "${data['displayName']}"');
    
    // Get the Firebase Auth user
    try {
      final userRecord = await auth.currentUser;
      if (userRecord != null && userRecord.uid == uid) {
        final correctEmail = userRecord.email?.toLowerCase() ?? '';
        print('  Email from Auth: "${userRecord.email}"');
        print('  Correct email (lowercase): "$correctEmail"');
        
        if (data['email'] != correctEmail) {
          print('  ⚠️ MISMATCH! Updating...');
          
          await firestore.collection('users').doc(uid).update({
            'email': correctEmail,
            'updatedAt': FieldValue.serverTimestamp(),
          });
          
          print('  ✅ Updated!');
        }
      }
    } catch (e) {
      print('  ❌ Error: $e');
    }
    print('');
  }
  
  print('✅ Done!');
}

File: fix_emails.js
// Quick script to fix user emails in Firestore
// Run with: node fix_emails.js

const admin = require('firebase-admin');

// Initialize Firebase Admin (uses application default credentials)
admin.initializeApp({
  projectId: 'tres3-5fdba'
});

const db = admin.firestore();

async function fixEmails() {
  try {
    // Get all users
    const usersSnapshot = await db.collection('users').get();
    
    console.log(`Found ${usersSnapshot.size} users`);
    
    for (const doc of usersSnapshot.docs) {
      const data = doc.data();
      const uid = doc.id;
      
      console.log(`\nUser ${uid}:`);
      console.log(`  Current email: "${data.email}"`);
      console.log(`  Display name: "${data.displayName}"`);
      
      // Get the Firebase Auth user to get the real email
      try {
        const userRecord = await admin.auth().getUser(uid);
        const correctEmail = userRecord.email ? userRecord.email.toLowerCase() : '';
        
        console.log(`  Auth email: "${userRecord.email}"`);
        console.log(`  Correct email (lowercase): "${correctEmail}"`);
        
        if (data.email !== correctEmail) {
          console.log(`  ⚠️  MISMATCH! Updating...`);
          
          await db.collection('users').doc(uid).update({
            email: correctEmail,
            updatedAt: admin.firestore.FieldValue.serverTimestamp()
          });
          
          console.log(`  ✅ Updated!`);
        } else {
          console.log(`  ✓ Email is correct`);
        }
      } catch (authError) {
        console.log(`  ❌ Error getting auth user: ${authError.message}`);
      }
    }
    
    console.log('\n✅ Done!');
  } catch (error) {
    console.error('Error:', error);
  }
}

fixEmails();
